lib/api/core/entrypoints/embedded/index.js

```diff
@@ -276,12 +282,15 @@ class EmbeddedEntryPoint extends EntryPoint {
    * @param {object} extra
    */
   logAccess (request, extra = null) {
-    const connection = this.clients[request.context.connection.id];
+    let connection = this.clients[request.context.connection.id];
```

Comment:
I wonder if we really need to get the whole `ClientConnection` object, and then use a fallback if the connecion is lost.

Can't we get all the information we need from `request.context.connection` object ?

Reply:
I mean we only need that (or I miss something ?):
```
const connection = {
  id: request.context.connection.id,
  protocol: request.context.connection.protocol || '-',
  ips: request.context.connection.ips || [},
  headers: request.context.connection.misc.headers || []
};
```

Reply:
I didn't consider that option :thinking: 
You're right, that shouldn't make a difference, neither data nor performance wise (at this point `ClientConnection` should be promoted to a pure C++ class by turbofan, and allocated on the stack, so the overhead should be comparable to getting a ref from the `clients` dictionnary).

I'll make the necessary changes. :+1:

Reply:
@ballinette > after extensive testing, there are cases when we have a connection but no `Request` associated to it. For instance, when we log an error about the HTTP query itself, before we can even parse it. So we do need a fallback.

Reply:
OK, let's keep this code as is ;)